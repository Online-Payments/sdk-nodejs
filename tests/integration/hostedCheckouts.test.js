const sdkInstance = require('./__fixture__');
const config = require('./__fixture__/config');
const stubs = require('./__stubs__');

describe('Hosted checkouts', () => {
  describe('Promises', () => {
    test('it can create a hosted checkout', async () => {
      const sdkResponse = await sdkInstance.hostedCheckout.createHostedCheckout(
        config.merchantId,
        stubs.createHostedCheckout,
        {}
      );
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });

    test('it can get a hosted checkout', async () => {
      const sdkResponse = await sdkInstance.hostedCheckout.getHostedCheckout(config.merchantId, 'hostedCheckoutId', {});
      expect(sdkResponse.isSuccess).toBe(true);
      expect(sdkResponse).toMatchSnapshot();
    });
  });

  describe('Callback', () => {
    test('it can create a hosted checkout', (done) => {
      sdkInstance.hostedCheckout.createHostedCheckout(
        config.merchantId,
        stubs.createHostedCheckout,
        {},
        (error, sdkResponse) => {
          expect(error).toBeNull();
          expect(sdkResponse.isSuccess).toBe(true);
          expect(sdkResponse).toMatchSnapshot();
          done();
        }
      );
    });

    test('it can get a hosted checkout', (done) => {
      sdkInstance.hostedCheckout.getHostedCheckout(config.merchantId, 'hostedCheckoutId', {}, (error, sdkResponse) => {
        expect(error).toBeNull();
        expect(sdkResponse.isSuccess).toBe(true);
        expect(sdkResponse).toMatchSnapshot();
        done();
      });
    });
  });
});
