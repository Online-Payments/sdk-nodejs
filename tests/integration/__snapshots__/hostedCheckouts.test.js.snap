// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hosted checkouts Callback it can create a hosted checkout 1`] = `
Object {
  "body": Object {
    "RETURNMAC": "fecab85c-9b0e-42ee-a9d9-ebb69b0c2eb0",
    "hostedCheckoutId": 3066019730,
    "invalidTokens": "['0a7042d6-bf24-40b6-85c5-dec24b9dd1d1', 'b7042dff-1fbd-46d2-9ecb-bddaa2c31165']",
    "merchantReference": "your-order-6372",
    "partialRedirectUrl": "string",
    "redirectUrl": "string",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Hosted checkouts Callback it can get a hosted checkout 1`] = `
Object {
  "body": Object {
    "createdPaymentOutput": Object {
      "payment": Object {
        "hostedCheckoutSpecificOutput": Object {
          "hostedCheckoutId": 3066019730,
          "variant": "my-custom-template.html",
        },
        "id": "3066019730_0",
        "paymentOutput": Object {
          "acquiredAmount": Object {
            "amount": 1000,
            "currencyCode": "EUR",
          },
          "amountOfMoney": Object {
            "amount": 1000,
            "currencyCode": "EUR",
          },
          "amountPaid": -9223372036854776000,
          "cardPaymentMethodSpecificOutput": Object {
            "authenticatedAmount": -9223372036854776000,
            "authorisationCode": "string",
            "card": Object {
              "bin": "string",
              "cardNumber": "string",
              "countryCode": "BE",
              "expiryDate": 529,
            },
            "externalTokenLinked": Object {
              "ComputedToken": "string",
              "GTSComputedToken": "string",
              "GeneratedToken": "string",
            },
            "fraudResults": Object {
              "avsResult": "string",
              "cvvResult": "string",
              "fraudServiceResult": "string",
            },
            "initialSchemeTransactionId": "string",
            "paymentOption": "string",
            "paymentProductId": 840,
            "schemeReferenceData": "string",
            "threeDSecureResults": Object {
              "acsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "appliedExemption": "low-value",
              "authenticationStatus": "Y",
              "cavv": "string",
              "challengeIndicator": "no-preference",
              "dsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "eci": "string",
              "exemptionEngineFlow": "not-applicable-challenge-indicator-received",
              "flow": "frictionless",
              "liability": "issuer",
              "schemeEci": 5,
              "version": "2.2.0",
              "xid": "string",
            },
            "token": "0ca037cc-9079-4df7-8f6f-f2a3443ee521",
          },
          "customer": Object {
            "device": Object {
              "ipAddressCountryCode": "BE",
            },
          },
          "merchantParameters": "SessionID=126548354&ShopperID=73541312",
          "mobilePaymentMethodSpecificOutput": Object {
            "authorisationCode": "string",
            "fraudResults": Object {
              "avsResult": "string",
              "cvvResult": "string",
              "fraudServiceResult": "string",
            },
            "network": "string",
            "paymentData": Object {
              "dpan": "string",
              "expiryDate": 529,
            },
            "paymentProductId": 840,
            "threeDSecureResults": Object {
              "acsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "appliedExemption": "low-value",
              "authenticationStatus": "Y",
              "cavv": "string",
              "challengeIndicator": "no-preference",
              "dsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "eci": "string",
              "exemptionEngineFlow": "not-applicable-challenge-indicator-received",
              "flow": "frictionless",
              "liability": "issuer",
              "schemeEci": 5,
              "version": "2.2.0",
              "xid": "string",
            },
          },
          "paymentMethod": "string",
          "redirectPaymentMethodSpecificOutput": Object {
            "fraudResults": Object {
              "fraudServiceResult": "string",
            },
            "paymentOption": "string",
            "paymentProduct5402SpecificOutput": Object {
              "brand": "string",
            },
            "paymentProduct5500SpecificOutput": Object {
              "paymentEndDate": "string",
              "paymentReference": "string",
              "paymentStartDate": "string",
            },
            "paymentProduct840SpecificOutput": Object {
              "billingAddress": Object {
                "additionalInfo": "floor 9",
                "city": "Zaventem",
                "countryCode": "BE",
                "houseNumber": 3,
                "state": "string",
                "street": "Da Vinci street",
                "zip": 1930,
              },
              "customerAccount": Object {
                "accountId": "customer-account@email.com",
                "companyName": "Customer Company Name",
                "countryCode": "BE",
                "customerAccountStatus": "verified",
                "customerAddressStatus": "confirmed",
                "firstName": "John",
                "payerId": "RRCYJUTFJGZTA",
                "surname": "Doe",
              },
              "customerAddress": Object {
                "additionalInfo": "floor 9",
                "city": "Zaventem",
                "countryCode": "BE",
                "houseNumber": 3,
                "state": "string",
                "street": "Da Vinci street",
                "zip": 1930,
              },
              "protectionEligibility": Object {
                "eligibility": "Eligible",
                "type": "ItemNotReceivedEligible",
              },
            },
            "paymentProductId": 840,
            "token": "0ca037cc-9079-4df7-8f6f-f2a3443ee521",
          },
          "references": Object {
            "merchantParameters": "SessionID=126548354&ShopperID=73541312",
            "merchantReference": "your-order-6372",
          },
          "sepaDirectDebitPaymentMethodSpecificOutput": Object {
            "fraudResults": Object {
              "fraudServiceResult": "string",
            },
            "paymentProduct771SpecificOutput": Object {
              "mandateReference": "string",
            },
            "paymentProductId": 840,
          },
          "surchargeSpecificOutput": Object {
            "mode": "pass-through",
            "surchargeAmount": Object {
              "amount": 1000,
              "currencyCode": "EUR",
            },
          },
        },
        "status": "CREATED",
        "statusOutput": Object {
          "errors": Array [
            Object {
              "category": "PAYMENT_PLATFORM_ERROR",
              "code": 50001130,
              "errorCode": 50001130,
              "httpStatusCode": 404,
              "id": "UNKNOWN_PAYMENT_ID",
              "message": "Authorisation declined",
              "propertyName": "paymentId",
              "retriable": true,
            },
          ],
          "isAuthorized": true,
          "isCancellable": true,
          "isRefundable": true,
          "statusCategory": "CREATED",
          "statusCode": -2147483648,
          "statusCodeChangeDateTime": "string",
        },
      },
      "paymentStatusCategory": "SUCCESSFUL",
    },
    "status": "PAYMENT_CREATED",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Hosted checkouts Promises it can create a hosted checkout 1`] = `
Object {
  "body": Object {
    "RETURNMAC": "fecab85c-9b0e-42ee-a9d9-ebb69b0c2eb0",
    "hostedCheckoutId": 3066019730,
    "invalidTokens": "['0a7042d6-bf24-40b6-85c5-dec24b9dd1d1', 'b7042dff-1fbd-46d2-9ecb-bddaa2c31165']",
    "merchantReference": "your-order-6372",
    "partialRedirectUrl": "string",
    "redirectUrl": "string",
  },
  "isSuccess": true,
  "status": 200,
}
`;

exports[`Hosted checkouts Promises it can get a hosted checkout 1`] = `
Object {
  "body": Object {
    "createdPaymentOutput": Object {
      "payment": Object {
        "hostedCheckoutSpecificOutput": Object {
          "hostedCheckoutId": 3066019730,
          "variant": "my-custom-template.html",
        },
        "id": "3066019730_0",
        "paymentOutput": Object {
          "acquiredAmount": Object {
            "amount": 1000,
            "currencyCode": "EUR",
          },
          "amountOfMoney": Object {
            "amount": 1000,
            "currencyCode": "EUR",
          },
          "amountPaid": -9223372036854776000,
          "cardPaymentMethodSpecificOutput": Object {
            "authenticatedAmount": -9223372036854776000,
            "authorisationCode": "string",
            "card": Object {
              "bin": "string",
              "cardNumber": "string",
              "countryCode": "BE",
              "expiryDate": 529,
            },
            "externalTokenLinked": Object {
              "ComputedToken": "string",
              "GTSComputedToken": "string",
              "GeneratedToken": "string",
            },
            "fraudResults": Object {
              "avsResult": "string",
              "cvvResult": "string",
              "fraudServiceResult": "string",
            },
            "initialSchemeTransactionId": "string",
            "paymentOption": "string",
            "paymentProductId": 840,
            "schemeReferenceData": "string",
            "threeDSecureResults": Object {
              "acsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "appliedExemption": "low-value",
              "authenticationStatus": "Y",
              "cavv": "string",
              "challengeIndicator": "no-preference",
              "dsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "eci": "string",
              "exemptionEngineFlow": "not-applicable-challenge-indicator-received",
              "flow": "frictionless",
              "liability": "issuer",
              "schemeEci": 5,
              "version": "2.2.0",
              "xid": "string",
            },
            "token": "0ca037cc-9079-4df7-8f6f-f2a3443ee521",
          },
          "customer": Object {
            "device": Object {
              "ipAddressCountryCode": "BE",
            },
          },
          "merchantParameters": "SessionID=126548354&ShopperID=73541312",
          "mobilePaymentMethodSpecificOutput": Object {
            "authorisationCode": "string",
            "fraudResults": Object {
              "avsResult": "string",
              "cvvResult": "string",
              "fraudServiceResult": "string",
            },
            "network": "string",
            "paymentData": Object {
              "dpan": "string",
              "expiryDate": 529,
            },
            "paymentProductId": 840,
            "threeDSecureResults": Object {
              "acsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "appliedExemption": "low-value",
              "authenticationStatus": "Y",
              "cavv": "string",
              "challengeIndicator": "no-preference",
              "dsTransactionId": "3E1D57DF-8DB1-4614-91D5-B11962519703",
              "eci": "string",
              "exemptionEngineFlow": "not-applicable-challenge-indicator-received",
              "flow": "frictionless",
              "liability": "issuer",
              "schemeEci": 5,
              "version": "2.2.0",
              "xid": "string",
            },
          },
          "paymentMethod": "string",
          "redirectPaymentMethodSpecificOutput": Object {
            "fraudResults": Object {
              "fraudServiceResult": "string",
            },
            "paymentOption": "string",
            "paymentProduct5402SpecificOutput": Object {
              "brand": "string",
            },
            "paymentProduct5500SpecificOutput": Object {
              "paymentEndDate": "string",
              "paymentReference": "string",
              "paymentStartDate": "string",
            },
            "paymentProduct840SpecificOutput": Object {
              "billingAddress": Object {
                "additionalInfo": "floor 9",
                "city": "Zaventem",
                "countryCode": "BE",
                "houseNumber": 3,
                "state": "string",
                "street": "Da Vinci street",
                "zip": 1930,
              },
              "customerAccount": Object {
                "accountId": "customer-account@email.com",
                "companyName": "Customer Company Name",
                "countryCode": "BE",
                "customerAccountStatus": "verified",
                "customerAddressStatus": "confirmed",
                "firstName": "John",
                "payerId": "RRCYJUTFJGZTA",
                "surname": "Doe",
              },
              "customerAddress": Object {
                "additionalInfo": "floor 9",
                "city": "Zaventem",
                "countryCode": "BE",
                "houseNumber": 3,
                "state": "string",
                "street": "Da Vinci street",
                "zip": 1930,
              },
              "protectionEligibility": Object {
                "eligibility": "Eligible",
                "type": "ItemNotReceivedEligible",
              },
            },
            "paymentProductId": 840,
            "token": "0ca037cc-9079-4df7-8f6f-f2a3443ee521",
          },
          "references": Object {
            "merchantParameters": "SessionID=126548354&ShopperID=73541312",
            "merchantReference": "your-order-6372",
          },
          "sepaDirectDebitPaymentMethodSpecificOutput": Object {
            "fraudResults": Object {
              "fraudServiceResult": "string",
            },
            "paymentProduct771SpecificOutput": Object {
              "mandateReference": "string",
            },
            "paymentProductId": 840,
          },
          "surchargeSpecificOutput": Object {
            "mode": "pass-through",
            "surchargeAmount": Object {
              "amount": 1000,
              "currencyCode": "EUR",
            },
          },
        },
        "status": "CREATED",
        "statusOutput": Object {
          "errors": Array [
            Object {
              "category": "PAYMENT_PLATFORM_ERROR",
              "code": 50001130,
              "errorCode": 50001130,
              "httpStatusCode": 404,
              "id": "UNKNOWN_PAYMENT_ID",
              "message": "Authorisation declined",
              "propertyName": "paymentId",
              "retriable": true,
            },
          ],
          "isAuthorized": true,
          "isCancellable": true,
          "isRefundable": true,
          "statusCategory": "CREATED",
          "statusCode": -2147483648,
          "statusCodeChangeDateTime": "string",
        },
      },
      "paymentStatusCategory": "SUCCESSFUL",
    },
    "status": "PAYMENT_CREATED",
  },
  "isSuccess": true,
  "status": 200,
}
`;
